### ISSUE TITLE
False positive of `redefined_variable_type` for `Mock` instances.

### ISSUE THREAD

**Issue by jonykalavera** [PyCQA/pylint#2546]
### Steps to reproduce
1. create the following files:
```python
# test1.py
# pylint: disable=missing-docstring
from unittest import TestCase, mock

class TestA(TestCase):
    def test_foo(self):
        my_mock = mock.Mock()
        my_mock.foo = 1
        self.assertEqual(my_mock.foo, 1)

```
```python
# test2.py
# pylint: disable=missing-docstring
from unittest import TestCase, mock


class TestB(TestCase):
    def test_foo(self):
        my_mock = mock.Mock()
        my_mock.foo.return_value = '1'
        self.assertEqual(my_mock.foo(), '1')
```
2. Run pylint like this:
```bash
$ pylint test1.py test2.py
************* Module test2
test2.py:9:25: E1102: my_mock.foo is not callable (not-callable)

------------------------------------------------------------------
Your code has been rated at 5.83/10 (previous run: 5.83/10, +0.00)
```
3. Run pylint like this

```bash
$ pylint test2.py test1.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```
### Current behavior

order affects the results

### Expected behavior

order should not affect the results

### pylint --version output
```bash
$ pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Sep 15 2018, 19:13:07)
[GCC 8.2.1 20180831]
```
